modules/affile/affile-source.c

```diff
@@ -104,20 +91,16 @@ affile_sd_init(LogPipe *s)
     return FALSE;
 
   file_reader_options_init(&self->file_reader_options, cfg, self->super.super.group);
+  if (!file_reader_options_validate(&self->file_reader_options))
```

Comment:
This feels more like the responsibility of the `FileReader` class, thus pushing it down the `FileReader` init function. What do you think ?

Reply:
I agree. This is here, because `wildcard-file()` creates its readers on-the-fly, so the validation would run after `init()`.

Reply:
We've discussed this IRL, `validate()` is not really necessary, I'll move the validation into `file_reader_options_init()` (changing its return value).